src/lib/crypto/kimchi_bindings/js/bindings.js

```diff
@@ -1703,49 +1703,13 @@ var PERMUTS_MINUS_1 = 6;
 // Provides: caml_pasta_fp_proof_evaluations_to_rust
 // Requires: plonk_wasm, caml_fp_vector_to_rust, PERMUTS_MINUS_1, COLUMNS
 var caml_pasta_fp_proof_evaluations_to_rust = function(x) {
-    var w = new plonk_wasm.WasmVecVecFp(COLUMNS);
-    for (var i = 0; i < COLUMNS; ++i) {
-      w.push(caml_fp_vector_to_rust(x[1][i + 1]));
-    }
-
-    var coefficients = new plonk_wasm.WasmVecVecFp(COLUMNS);
```

Comment:
perhaps in the `WasmVecVec` file it would be nice to have the explanations you gave me. Perhaps something like

```
//! Accessing Rust vector values is not the same as accessing an array.
//! Each indexing (e.g. `some_vec[3]`) is costly as it is implemented as a function call.
//! Knowing that, plus the fact that field elements are implemented as `[u64; 8]`, we know that we incur the cost of following several pointers.
//! To decrease that cost, we flatten such arrays, going from something like
//!
//! ```
//! [[a, b], [c, d], ...]
//! ```
//!
//! to a flattened vector like:
//!
//! ```
//! [a, b, c, d, ...]
```